src/master-playlist-controller.js

```diff
@@ -408,6 +408,19 @@ export class MasterPlaylistController extends videojs.EventTarget {
         bubbles: true
       });
     });
+
+    this.masterPlaylistLoader_.on('playlistunchanged', () => {
+      let updatedPlaylist = this.masterPlaylistLoader_.media();
+      let playlistOutdated = this.stuckAtPlaylistEnd_(updatedPlaylist);
+
+      // when the playlist is unchanged and the playhead is stuck at the edge
+      // of playlist, we will blacklist the outdated playlist if it's not the final
+      // rendition to give the player a chance to re-adjust to the proper live point
+      // on a different playlist
+      if (playlistOutdated) {
+        this.blacklistCurrentPlaylist();
```

Comment:
Thinking again about the comment, I think it might make sense to put it in here as something like:

```js
// Playlist has stopped updating and we're stuck at its end. Try to blacklist it and switch to another
// playlist in the hope that that one is updating (and give the player a chance to re-adjust to the
// safe live point).
```